lib/std/mem.zig

```diff
@@ -424,6 +424,9 @@ test "zeroes" {
 
     comptime var comptime_union = zeroes(C_union);
     try testing.expectEqual(@as(u8, 0), comptime_union.a);
+
+    // https://github.com/ziglang/zig/pull/12246
```

Comment:
1. I dont understand what this comment is for.
2. Do you think it would make sense to also add test coverage for the recursive struct case (struct_info.layout == .Extern and the other branch)?

Reply:
Oh I was just thinking it motivates the test because it otherwise looks useless...